ArchiSteamFarm/IPC/Startup.cs

```diff
@@ -97,44 +100,41 @@ internal sealed class Startup {
 		// Add support for default root path redirection (GET / -> GET /index.html), must come before static files
 		app.UseDefaultFiles();
 
-		// Add support for static files (e.g. HTML, CSS and JS from IPC GUI)
-		app.UseStaticFiles(
-			new StaticFileOptions {
-				OnPrepareResponse = static context => {
-					if (context.File is { Exists: true, IsDirectory: false } && !string.IsNullOrEmpty(context.File.Name)) {
-						string extension = Path.GetExtension(context.File.Name);
-
-						CacheControlHeaderValue cacheControl = new();
-
-						switch (extension.ToUpperInvariant()) {
-							case ".CSS" or ".JS":
-								// Add support for SRI-protected static files
-								// SRI requires from us to notify the caller (especially proxy) to avoid modifying the data
-								cacheControl.NoTransform = true;
-
-								goto default;
-							default:
-								// Instruct the caller to always ask us first about every file it requests
-								// Contrary to the name, this doesn't prevent client from caching, but rather informs it that it must verify with us first that his cache is still up-to-date
-								// This is used to handle ASF and user updates to WWW root, we don't want from the client to ever use outdated scripts
-								cacheControl.NoCache = true;
-
-								// All static files are public by definition, we don't have any authorization here
-								cacheControl.Public = true;
-
-								break;
-						}
+#if !NETFRAMEWORK && !NETSTANDARD
+		Dictionary<string, string> pluginsPaths = new();
+
+		if (PluginsCore.ActivePlugins?.Count > 0) {
+			foreach (IWebInterface plugin in PluginsCore.ActivePlugins.OfType<IWebInterface>()) {
+				if (string.IsNullOrEmpty(plugin.PhysicalPath) || string.IsNullOrEmpty(plugin.WebPath)) {
+					continue;
+				}
+
+				string staticFilesDirectory = Path.Combine(Path.GetDirectoryName(plugin.GetType().Assembly.Location)!, plugin.PhysicalPath);
 
-						ResponseHeaders headers = context.Context.Response.GetTypedHeaders();
+				if (Directory.Exists(staticFilesDirectory)) {
+					pluginsPaths.Add(staticFilesDirectory, plugin.WebPath);
 
-						headers.CacheControl = cacheControl;
+					if (plugin.WebPath != "/") {
+						app.UseDefaultFiles(plugin.WebPath);
 					}
 				}
 			}
-		);
+		}
+
+		// Add support for static files from custom plugins (e.g. HTML, CSS and JS)
+		foreach (KeyValuePair<string, string> pluginPaths in pluginsPaths) {
```

Comment:
We can move the body of this loop up into the above loop and remove allocation of `pluginsPaths`. If we keep this, I'd still recommend deconstructing the kvp as such:
```csharp
foreach((string fileSystemPath, string requestPath) in pluginPaths) {
    // ...
}
```

Reply:
> We can move the body of this loop up into the above loop and remove allocation of `pluginsPaths`. If we keep this, I'd still recommend deconstructing the kvp as such:
> 
> ```cs
> foreach((string fileSystemPath, string requestPath) in pluginPaths) {
>     // ...
> }
> ```

nope we can't, acording to docs and archi comments: Default files must come before static files.

Reply:
Misc, not a blocker for this PR.